docs: Install -- SAML (#4164)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.1/security/howto-dex-saml-connector/index.md

From step 2 you need:

- issuer URL -> `entityIssuer`
- callback URL -> `redirectURI`

Make sure you base64 encode the contents of the PEM file. As an example, the prefix of the contents will result into this exact base64 prefix:

```text
LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tC[...]
```

Update your `dex` addon configuration:

```yaml
    - name: dex
      enabled: true
      values: |
        config:
          connectors:
          - type: saml
            id: saml
            name: SAML
            config:
              ssoURL: <url for POST request>
              caData: < base64 PEM encoded CA for the IdP server >
              redirectURI: https://<your-cluster-host>/dex/callback
              entityIssuer: https://<your-cluster-host>/dex
              usernameAttr: < user attribute in saml response >
              emailAttr: < email attribute in saml response >
```

Step 4: Modify `traefik-foward-auth` addon in your `cluster.yaml` and add a whitelist

This step is required to give access to a user to ops-portal. For each user, you must [give access to Kubernetes resources](../rbac) and add an entry in the `whitelist` below.

```yaml

>    - name: traefik-forward-auth
>      enabled: true
>      values: |
>        traefikForwardAuth:
>          allowedUser:
>            valueFrom:
>              secretKeyRef: null
>          whitelist:
>          - < allowed email addresses >

Reviewer: ```suggestion
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  traefik-forward-auth-mgmt:
    values: |
      config:
        traefikForwardAuth:
          allowedUser:
            valueFrom:
              secretKeyRef: null
          whitelist:
          - < allowed email addresses >
[...]
```

```

> Step 5: Run `Kommander up` to deploy modified Dex.

Reviewer: ```suggestion
Step 5: Run `kommander install` to deploy modified `dex`.
```

> Step 6: Visit `https://<your-cluster-host>/ops/landing` to login to ops portal.

Reviewer: ```suggestion
Step 6: Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.
```

---
layout: layout.pug
navigationTitle: SAML
title: SAML connector
menuWeight: 33
excerpt: Connect your Kommander cluster to an Idp using SAML
beta: false
enterprise: false
---

> ## How to connect Kommander to an IdP using SAML

Reviewer 2: ```suggestion
## Connect Kommander to an IdP using SAML
```

> This guide shows how to configure your Kommander cluster to use SAML to connect to an identity provider (IdP).

Reviewer 2: ```suggestion
This section shows how to configure your Kommander cluster to use SAML, so that you can connect to an identity provider (IdP).
```

> Step 1: [Install Kommander](../../install).

Reviewer 2: ```suggestion
1.   [Install Kommander](../../install).
```

> Step 2: Configure the IdP

Reviewer 2: ```suggestion
1.   Configure the IdP
```

Author: Just to double check, did you mean `2.` here instead of `1.`?

Author: Actually, I've just noticed that all the steps have been changed to `1.`. Can I ask why that is?

Reviewer: In markdown you can number all steps with `1.`'s which automatically renders as an ordered list, which is nice so if you ever had to add a step in the middle you don't need to redo the entire numbered list https://www.markdownguide.org/basic-syntax/#ordered-lists

Author: Good to know! However, locally, it's not working as expected. The first two steps are numbered correctly, but the 3rd step is resetting back to 1. I'm guessing because of all the copy in between the steps. Not sure if there's a way around this or if we need to go with a different approach.

Reviewer: Probably something to do with the indents and spacing to get it working properly. I can take a look at your branch to try to get it working

Author: That's alright, I'm sure you've got loads of your own work! You've already been super helpful. I can play around with it on Monday and see if I sort it out. Thanks for the offer though!

Reviewer 3: @seialkali, could you please let me know if you find a solution? I am currently working on a document where I am seeing the same behavior.

> You will provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP.

Reviewer 2: ```suggestion
Provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP.
```

> You will provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP.
> 
> The issuer URL points to the authentication endpoint at the service provider (Dex) that issues a request towards the IdP via the user agent.

Reviewer 2: ```suggestion
Provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP. The issuer URL points to the authentication endpoint at the service provider (Dex), which issues a request towards the IdP via the user agent.
```

The issuer URL follows this schema:

```text
https://<your-cluster-host>/dex
```

> The ACS URL points to the service provider (Dex) endpoint that will receive SAML assertions issued by the IdP.

Reviewer 2: ```suggestion
The ACS URL points to the service provider (Dex) endpoint that receives SAML assertions issued by the IdP.
```

> The ACS or callback URL will look like this:

Reviewer 2: ```suggestion
The ACS or callback URL should look like this:
```

```text
https://<your-cluster-host>/dex/callback
```

> Depending on the IdP, you may be asked to provide the configuration in some form of an XML snippet - see an example below;

Reviewer 2: ```suggestion
Depending on the IdP, you might be asked to provide the configuration in some form of an XML snippet - see the following example:
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<EntityDescriptor xmlns="urn:oasis:names:tc:SAML:2.0:metadata" entityID="https://<your-cluster-host>/dex">
   <SPSSODescriptor AuthnRequestsSigned="false" WantAssertionsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
      <NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</NameIDFormat>
      <AssertionConsumerService index="0" isDefault="true" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://<your-cluster-host>/dex/callback" />
   </SPSSODescriptor>
</EntityDescriptor>
```

> Step 3: Modify the `dex` configuration:

Reviewer 2: ```suggestion
1.   Modify the `dex` configuration:
```

> For this step, you will need to get the following from your IdP:

Reviewer 2: ```suggestion
For this step, obtain the following from your IdP:
```

- single sign-on URL or SAML URL -> `ssoURL`
- base64 encoded, PEM encoded CA certificate -> `caData`
- username attribute name in SAML response -> `usernameAttr`
- email attribute name in SAML response -> `emailAttr`

From step 2 you need:

- issuer URL -> `entityIssuer`
- callback URL -> `redirectURI`

> Make sure you base64 encode the contents of the PEM file. As an example, the prefix of the contents will result into this exact base64 prefix:

Reviewer 2: ```suggestion
Ensure you base64 encode the contents of the PEM file. As an example, the prefix of the contents will result into this exact base64 prefix:
```

```text
LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tC[...]
```

> You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the Kommander CLI’s install command would look similar to this:

Reviewer 2: ```suggestion
You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the Kommander CLI’s install command should look similar to:
```

```yaml
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  dex:
    values: |
      config:
        connectors:
        - type: saml
          id: saml
          name: SAML
          config:

>             ssoURL: <url for POST request>

Reviewer: ```suggestion
            ssoURL: < url for POST request >
```
tiny nit to make it consistent with the others

            caData: < base64 PEM encoded CA for the IdP server >
            redirectURI: https://<your-cluster-host>/dex/callback
            entityIssuer: https://<your-cluster-host>/dex
            usernameAttr: < user attribute in saml response >
            emailAttr: < email attribute in saml response >
[...]
```

> Step 4: Modify the `traefik-foward-auth-mgmt` configuration and add a whitelist:

Reviewer 2: ```suggestion
1.   Modify the `traefik-foward-auth-mgmt` configuration and add a whitelist:
```

> This step is required to give access to a user to ops-portal. For each user, you must [give access to Kubernetes resources](../rbac) and add an entry in the `whitelist` below.

Reviewer: ```suggestion
This step is required to give access to a user to the Kommander dashboard. For each user, you must [give access to Kubernetes resources](../rbac) and add an entry in the `whitelist` below.
```

Should we be calling the UI "Kommander dashboard" or something like "web interface" ? (I've seen both)

Author: I can ask in the dkp-ux channel

Author: @gvasquezvargas should I just use `Kommander UI` here?

Reviewer 3: I guess it depends. Is the UI's wording already been changed to DKP? If yes, I would rather change it to DKP UI already, but if it still called Kommander, I would leave it that way. @cmastr what do you think?

Author: Changing this to `DKP UI` based on this Slack thread, but if this is wrong for some reason, let me know: https://mesosphere.slack.com/archives/C052HL47U/p1647015731820569

```yaml
apiVersion: config.kommander.mesosphere.io/v1alpha1
kind: Installation
apps:
  traefik-forward-auth-mgmt:
    values: |
      traefikForwardAuth:
        allowedUser:
          valueFrom:
            secretKeyRef: null
        whitelist:
        - < allowed email addresses >
[...]
```

> Step 5: Run `kommander install` to deploy modified `dex`.

Reviewer 2: ```suggestion
1.   Run `kommander install` to deploy modified `dex`.
```

> Step 6: Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.
> 
> Step 7: Select `Launch Console` and follow the authentication steps.

Reviewer 2: ```suggestion
1.   Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.

1.   Select `Launch Console` and follow the authentication steps to complete the procedure.
```

---
layout: layout.pug
navigationTitle: SAML
title: SAML connector
menuWeight: 33
excerpt: Connect your Kommander cluster to an IdP using SAML
beta: false
enterprise: false
---

## Connect Kommander to an IdP using SAML

This procedure configures your Kommander cluster to use SAML, to connect to an identity provider (IdP).

1.  [Install Kommander](../../install).

1.  Configure the IdP

    Provide the issuer URL and the Assertion Consumer Service (ACS) or callback URL to your IdP. The issuer URL points to the authentication endpoint at the service provider (Dex), which issues a request towards the IdP via the user agent.

    The issuer URL follows this schema:

    ```text
    https://<your-cluster-host>/dex
    ```

    The ACS URL points to the service provider (Dex) endpoint that receives SAML assertions issued by the IdP.

    The ACS or callback URL should look like this:

    ```text
    https://<your-cluster-host>/dex/callback
    ```

>     Depending on the IdP, you might be asked to provide the configuration in some form of an XML snippet. See the following example, making sure to replace `<your-cluster-host>` with your URL:

Reviewer 4: ```suggestion
    Depending on the IdP, you may need to provide the configuration in an XML snippet. See the following example, making sure to replace `<your-cluster-host>` with your URL:
```

    ```xml
    <?xml version="1.0" encoding="UTF-8"?>
    <EntityDescriptor xmlns="urn:oasis:names:tc:SAML:2.0:metadata" entityID="https://<your-cluster-host>/dex">
        <SPSSODescriptor AuthnRequestsSigned="false" WantAssertionsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
            <NameIDFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</NameIDFormat>
            <AssertionConsumerService index="0" isDefault="true" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://<your-cluster-host>/dex/callback" />
        </SPSSODescriptor>
    </EntityDescriptor>
    ```

> 1.  Modify the `dex` configuration:

Reviewer 4: ```suggestion
1.  Edit the `dex` configuration:
```

    For this step, obtain the following from your IdP:

    - single sign-on URL or SAML URL -> `ssoURL`
    - base64 encoded, PEM encoded CA certificate -> `caData`
    - username attribute name in SAML response -> `usernameAttr`
    - email attribute name in SAML response -> `emailAttr`

    From above you need:

    - issuer URL -> `entityIssuer`
    - callback URL -> `redirectURI`

    Ensure you base64 encode the contents of the PEM file. As an example, the prefix of the contents will result into this exact base64 prefix:

    ```text
    LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tC[...]
    ```

>     You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the [Kommander CLI’s install command][kommander-install] should look similar to:

Reviewer 4: ```suggestion
    You can add the configuration as the values field in the `dex` application. An example `dex` configuration provided to the [Kommander CLI’s install command][kommander-install] looks like:
```

    ```yaml
    apiVersion: config.kommander.mesosphere.io/v1alpha1
    kind: Installation
    apps:
      dex:
        values: |
          config:
          connectors:
          - type: saml
              id: saml
              name: SAML
              config:
                ssoURL: < url for POST request >
                caData: < base64 PEM encoded CA for the IdP server >
                redirectURI: https://<your-cluster-host>/dex/callback
                entityIssuer: https://<your-cluster-host>/dex
                usernameAttr: < user attribute in saml response >
                emailAttr: < email attribute in saml response >
    [...]
    ```

> 1.  Modify the `traefik-foward-auth-mgmt` configuration and add a whitelist:

Reviewer 4: ```suggestion
1.  Edit the `traefik-foward-auth-mgmt` configuration and add a whitelist:
```

    This step is required to give access to a user to the Kommander dashboard. For each user, you must [give access to Kubernetes resources](../../operations/access-control/rbac) and add an entry in the `whitelist` below.

    ```yaml
    apiVersion: config.kommander.mesosphere.io/v1alpha1
    kind: Installation
    apps:
    ...
      traefik-forward-auth-mgmt:
        values: |
          traefikForwardAuth:
            allowedUser:
              valueFrom:
                secretKeyRef: null
            whitelist:
            - < allowed email addresses >

    ```

1.  Run `kommander install --installer-config kommander.yaml` to deploy modified `dex`.

1.  Visit `https://<your-cluster-host>/dkp/kommander/dashboard` to login to the Kommander dashboard.

1.  Select `Launch Console` and follow the authentication steps to complete the procedure.

> [kommander-install]: ../../install/configuration/

Author: @msdolbey how come this has been add to the bottom of the page?

Reviewer 4: Brittany this is the way we do links. So the link above in the sentence links to the bottom of the page, then the bottom of the page links to the actual page. That way we can reuse links throughout a document without duplicating efforts.

Reviewer 4: I see now there are some incorrectly formatted links in this document.
